iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -560,47 +566,10 @@ SmallVector<Range> FftOp::getLoopBounds(OpBuilder &builder) {
   return res;
 }
 
-// Generates FFT stage scalar implementation. This follows Cooley–Tukey FFT
-// algorithm. The pseudo reference code is:
-//   let s <- stage of linalg_ext.fft
-//   int m = 1 << s;
-//   int mh = m >> 1;
-//   for (int k = 0; k < n; k += m) {
-//     for (int j = 0; j < mh; ++j) {
-//       cplx w = exp(-2 * PI * j / m * I);
-//       cplx t = w * a[k + j + mh];
-//       cplx u = a[k + j];
-//       a[k + j] = u + t;
-//       a[k + j + mh] = u - t;
-//     }
-//   }
-LogicalResult FftOp::generateScalarImplementation(OpBuilder &b, Location loc,
-                                                  ValueRange ivs) {
-  Value real = getReal();
-  Value imag = getImag();
-  Value stage = getStage();
-  Value one = b.create<ConstantIndexOp>(loc, 1);
-  Value wholeSize = b.create<ShiftLeftOp>(loc, one, stage);
-  Value halfSize = b.create<SignedShiftRightOp>(loc, wholeSize, one);
-
+void FftOp::generateScalarImplWithoutCoeffBuf(OpBuilder &b, Location loc,
```

Comment:
Confused by the naming here. This seems to be the implementation when the coefficient buffer is passed as argument, and below implementation  seems to be one with coefficient buffer not passed in.

Reply:
There are two methods. I refactored the old implementation to `generateScalarImplWithoutCoeffBuf`.

I implemented another method `generateScalarImplWithCoeffBuf` below which implements the coefficient buffer is passed as arguments (at l.625 to l.664).